### PR TITLE
Vue Support: Setup deployment of Vue example app and add Vue docs

### PR DESCRIPTION
**Pull request by philippd**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [x] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description
Setup the deployment of the Vue demo app and add documentation for the Vue library.

This PR also includes a small bug fix: It fixes an issue where the Vue plugin throws an error if no options are provided during registration. Additionally, it renames the Vue plugin to `UiLibraryVue` for better consistency with the Angular version.

The upcoming section now also describes Vue support no longer as "experimental", as I believe it is stable enough now that v-model and the router is working as expected. 

### 📝 Checklist
- [x] It's submitted to the `main` branch
- [x] I have updated the documentation accordingly.
- [x] All tests are passing
- [x] I have updated the "upcoming" section inside _docs/changelog.md_ explaining the changes I contributed

### CODE REVIEW

File: docs/changelog.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).
 
 ### Added
 
-- Experimental support for Vue
+- Support for Vue
```
Comment (pennal) [inline docs/changelog.md:11]: Initial/Beta Support for Vue

File: docs/guide/vue.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
+# Vue
```
Comment (pennal) [inline docs/guide/vue.md:1]: As discussed during the weekly meeeting, please add a banner/alert stating that support for vue is in its infancy

File: docs/guide/vue.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+## Setup
+
+1. Install both the web components and Vue library:
+
+   ```bash
+   npm install @six-group/ui-library
+   npm install @six-group/ui-library-vue
+   ```
+
+2. Add the `UiLibraryVue` plugin to your Vue application and import the styles:
```
Comment (pennal) [inline docs/guide/vue.md:19]: In your main.ts file, add the ....

File: docs/index.md [hunk 4]
```diff
@@ -13,11 +13,14 @@ hero:
...
       link: https://six-group.github.io/six-webcomponents/demo/angular
+    - theme: alt
+      text: Vue Example
+      link: https://six-group.github.io/six-webcomponents/demo/vue
 features:
   - title: Agnostic
     details:
-      Our Library has first class support for Angular and React right out of the box. Checkout the
-      example apps above!
+      Our Library has first class support for Angular, React and Vue right out of the box. Checkout
```
Comment (pennal) [inline docs/index.md:22]: I would refrain from stating that we have full support for Vue. I would split Vue outside and say "beta support for Vue"

**Review by pennal — DISMISSED** [review]
I am approving this change on the basis that you will implement the fixes I suggested. But use your best judgement, if you think they are not needed, I am okay with it

**Review by ibirrer — APPROVED** [review]
I believe Vue support is mature enough and I don't think that we need to label it as beta.

**Comment by philippd** [comment]
@ibirrer I agree, we have no issues with the current Vue support in the 2 projects that are already using many of the components.

**Review by ibirrer — APPROVED** [review]

**Pull request merged**